internal/services/signalr/signalr_service_resource.go

```diff
@@ -468,6 +469,45 @@ func signalRFeature(featureFlag signalr.FeatureFlags, value string) signalr.Sign
 	}
 }
 
+func expandUpstreamAuth(input []interface{}) *signalr.UpstreamAuthSettings {
+	if len(input) == 0 {
+		return nil
+	}
+
+	result := signalr.UpstreamAuthSettings{}
+
+	inputRaw := input[0].(map[string]interface{})
+
+	authType := signalr.UpstreamAuthType(inputRaw["type"].(string))
+	result.Type = &authType
+
+	if inputRaw["managed_identity_resource_id"] != "" {
+		managedIdentityId := inputRaw["managed_identity_resource_id"].(string)
+		result.ManagedIdentity = &signalr.ManagedIdentitySettings{
+			Resource: utils.String(managedIdentityId),
+		}
+	}
+
+	return &result
+}
+
+func flattenUpstreamAuth(auth *signalr.UpstreamAuthSettings) []interface{} {
+	result := make([]interface{}, 0)
+	if auth == nil {
```

Comment:
we also want to set an empty list into the state if `type` isn't set to `ManagedIdentity`, so we can make this:

```suggestion
	if auth == nil || auth.Type == nil || *auth.Type != signalr.UpstreamAuthTypeManagedIdentity {
```